web/src/components/modals/add-view-options.tsx

```diff
@@ -0,0 +1,184 @@
+import { FC, useState } from 'react';
+import { Modal, Tab, Tabs } from 'react-bootstrap';
+import ReactSelect from 'react-select';
+import { Controller, FieldErrors, useForm } from 'react-hook-form';
+import { ErrorMessage } from '@hookform/error-message';
+
+
+import { useViewMutations } from '../../hooks/mutations/useViewMutations';
+import { useProjectPage } from '../../contexts/project-page-context';
+import { CreateProjectViewRequest, addProjectView, deleteProjectView } from '../../api/project';
+import { useProjectViews } from '../../hooks/queries/useProjectViews';
+import { useProjectSelectedView } from '../../hooks/stores/useProjectSelectedViewStore';
+
+interface Props {
+  show: boolean;
+  onHide: () => void;
+  filteredSamples: string[];
+}
+
+export const ViewOptionsModal: FC<Props> = ({ show, onHide, filteredSamples}) => {
```

Comment:
Here too... I don't know where I picked up this pattern, but I think its less clear... The `FC<Props>` is weird:

You can just make it:
`export const ViewOptionsModal = (props: Props) => { }`
Then restructure the props in the component